### PR TITLE
Update AnalysisFromFunction results to fit version 2.0.0

### PR DESCRIPTION
**Pull request by FloLangenfeld**
As per version 2.0.0 of MDAnalysis, the results generated with the `AnalysisFromFunction` class are now stored in `results.timeseries`.

This PR updates the User Guide to reflect this change.

A companion PR ([#3413](https://github.com/MDAnalysis/mdanalysis/pull/3413) in the MDAnalysis repository) does exist for the MDAnalysis code documentation. For consistency between code documentation and the User Guide, I kept the same syntax, but it could be easily changed, if required.

Changes made in this Pull Request:

 - Update use cases of AnalysisFromFunction to match MDAnalysis v2.0.0 API

### CODE REVIEW

**Comment by lilyminium** [comment]
Hi @FloLangenfeld, thank you for opening this PR! What you have already looks good to me, just a few small comments:

- Please add yourself to AUTHORS in `doc/source`!
- Could you please update the versions in the notebooks you've updated to 2.0.0 (in each top-level cell there should be a **Minimum version** specification)
- It would be great if you modified the custom analysis notebook. The first time where `results` is introduced is here:

```
Running the analysis iterates over the trajectory. The output is saved in rog.results, which has the same number of rows, as frames in the trajectory.
```
This should probably get updated to say "the output is saved in `rog.results.timeseries`, and a brief example showing that you can access the results both at `rog.results.timeseries` and `rog.results['timeseries']` would be great!

File: doc/source/examples/analysis/custom_trajectory_analysis.ipynb [hunk 1]
```diff
@@ -191,7 +172,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Running the analysis iterates over the trajectory. The output is saved in `rog.results`, which has the same number of rows, as frames in the trajectory."
+    "Running the analysis iterates over the trajectory. The output is saved in `rog.results`, which has the same number of rows, as frames in the trajectory. The output is saved in rog.results.timeseries. Below, a brief example shows that you can access the results both at rog.results.timeseries and rog.results['timeseries']."
```
Comment (lilyminium) [inline doc/source/examples/analysis/custom_trajectory_analysis.ipynb:175]:
```suggestion
    "Running the analysis iterates over the trajectory. The output is saved in `rog.results.timeseries`, which has the same number of rows, as frames in the trajectory. You can access the results both at `rog.results.timeseries` and `rog.results['timeseries'].`"
```

Could you please add a code cell below this, showing both, please? Currently the cell below only shows `rog.results['timeseries']`.

**Review by lilyminium — CHANGES_REQUESTED** [review]

File: doc/source/examples/analysis/custom_trajectory_analysis.ipynb (hunk 1 above)
Comment (FloLangenfeld) [reply doc/source/examples/analysis/custom_trajectory_analysis.ipynb]: Done. Thanks for the comment!

**Review by lilyminium — APPROVED** [review]

**Comment by lilyminium** [comment]
Thanks so much for this fix @FloLangenfeld!

**Comment by FloLangenfeld** [comment]
Happy to help!

Thank you @lilyminium for the helpful comments and remarks on this PR 👍🏽

**Pull request merged**